dev-packages/cli/src/run-test.ts

```diff
@@ -40,105 +31,36 @@ export default async function runTest(options: TestOptions): Promise<void> {
-            window.addEventListener('beforeunload', () => (window as any)['fireDidUnloadTheia']())
-        );
-    };
-    page.on('frameattached', preLoad);
-    page.on('framenavigated', preLoad);
-
-    page.on('load', async () => {
-        console.log('loading mocha...');
-        // replace console.log by theia logger for mocha
-        await page.waitForFunction(() => !!(window as any)['theia']['@theia/core/lib/common/logger'].logger, {
-            timeout: 30 * 1000
-        });
-        await page.addScriptTag({ path: require.resolve('mocha/mocha.js') });
-        await page.waitForFunction(() => !!(window as any)['chai'] && !!(window as any)['mocha'] && !!(window as any)['theia'].container, { timeout: 30 * 1000 });
-
-        console.log('loading Theia...');
-        await page.evaluate(() => {
-            const { FrontendApplicationStateService } = (window as any)['theia']['@theia/core/lib/browser/frontend-application-state'];
-            const { PreferenceService } = (window as any)['theia']['@theia/core/lib/browser/preferences/preference-service'];
-            const { WorkspaceService } = (window as any)['theia']['@theia/workspace/lib/browser/workspace-service'];
-
-            const container = (window as any)['theia'].container;
-            const frontendApplicationState = container.get(FrontendApplicationStateService);
-            const preferenceService = container.get(PreferenceService);
-            const workspaceService = container.get(WorkspaceService);
-
-            return Promise.all([
-                frontendApplicationState.reachedState('ready'),
-                preferenceService.ready,
-                workspaceService.roots
-            ]);
-        });
-
-        console.log('loading test files...');
-        await page.evaluate(() => {
-            // replace require to load modules from theia namespace
-            (window as any)['require'] = (moduleName: string) => (window as any)['theia'][moduleName];
-            mocha.setup({
-                reporter: 'spec',
-                ui: 'bdd',
-                useColors: true
-            } as MochaSetupOptions);
-        });
-        const files = collectFiles(fileOptions);
-        for (const file of files) {
-            await page.addScriptTag({ path: file });
-        }
-
-        console.log('running test files...');
-        const failures = await page.evaluate(() =>
-            new Promise<number>(resolve => mocha.run(resolve))
-        );
-        if (options.coverage) {
-            console.log('collecting test coverage...');
-            const [jsCoverage, cssCoverage] = await Promise.all([
-                page.coverage.stopJSCoverage(),
-                page.coverage.stopCSSCoverage(),
-            ]);
-            require('puppeteer-to-istanbul').write([...jsCoverage, ...cssCoverage]);
-        }
-        if (exit) {
-            await page.close();
-            process.exit(failures > 0 ? 1 : 0);
+    const testPage = await newTestPage({
```

Comment:
@vince-fugnitto you will need something similar to test theia-apps, but it should spawn a server in the docker container